apps/src/maze/MazeVisualizationColumn.jsx

```diff
@@ -0,0 +1,29 @@
+var msg = require('../locale');
+var commonStyles = require('../commonStyles');
+
+var Visualization = require('./Visualization');
+var StepButton = require('./StepButton');
+var SpellingControls = require('./SpellingControls');
+var GameButtons = require('../templates/GameButtons');
+var BelowVisualization = require('../templates/BelowVisualization');
+
+var MazeVisualizationColumn = function (props) {
+  return (
+    <span>
+      <Visualization/>
+      <GameButtons hideRunButton={props.hideRunButton}>
+        <StepButton showStepButton={props.showStepButton}/>
+      </GameButtons>
+      {props.searchWord && <SpellingControls searchWord={props.searchWord}/>}
+      <BelowVisualization/>
+    </span>
+  );
+};
```

Comment:
To start with, each app will end up with their own one of these. They will all have a pretty common structure (Visualzation, that is sometimes empty, GameButtons with additional per-app buttons, potentially some other stuff, then BelowVisualization which is the same everywhere - except that Calc sticks some other contents in there).

It might make sense to unify these with a single VisualizationColumn that takes as inputs the parts that change, but I'm going to start out un-unified.